plugins/Jobs.cpp

```diff
@@ -165,33 +165,89 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
-        // are processed
-        SINFO("Job already existed and unique flag was passed, reusing existing job " << result[0][0]);
-        content["jobID"] = result[0][0];
-        return true;
+        for (auto& job : jsonJobs) {
+            // If no priority set, set it
+            int64_t priority = SContains(job, "priority") ? SToInt(job["priority"]) : JOBS_DEFAULT_PRIORITY;
+
+            // We'd initially intended for any value to be allowable here, but for
+            // performance reasons, we currently will only allow specific values to
+            // try and keep queries fast. If you pass an invalid value, we'll throw
+            // here so that the caller can know that he did something wrong rather
+            // than having his job sit unprocessed in the queue forever. Hopefully
+            // we can remove this restriction in the future.
+            if (priority != 0 && priority != 500 && priority != 1000) {
+                throw "402 Invalid priority value";
+            }
+
+            // Throw if data is not a valid JSON object, otherwise UPDATE query will fail.
+            if (SContains(job, "data") && SParseJSONObject(job["data"]).empty()) {
+                throw "402 Data is not a valid JSON Object";
+            }
+
+            // Validate that the parentJobID exists and is in the right state if one was passed.
+            int64_t parentJobID = SContains(job, "parentJobID") ? SToInt(job["parentJobID"]) : 0;
+            if (parentJobID) {
+                SQResult result;
+                if (!db.read("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";", result)) {
+                    throw "502 Select failed";
+                }
+                if (result.empty()) {
+                    throw "404 parentJobID does not exist";
+                }
+                if (!SIEquals(result[0][0], "RUNNING") && !SIEquals(result[0][0], "PAUSED")) {
+                    SWARN("Trying to create child job with parent jobID#" << parentJobID << ", but parent isn't RUNNING or PAUSED (" << result[0][0] << ")");
+                    throw "405 Can only create child job when parent is RUNNING or PAUSED";
+                }
+            }
+
+            // Verify unique
+            SQResult result;
+            SINFO("Unique flag was passed, checking existing job with name " << job["name"]);
+            if (!db.read("SELECT jobID, data "
+                         "FROM jobs "
+                         "WHERE name=" + SQ(job["name"]) + ";",
+                         result)) {
+                throw "502 Select failed";
+            }
+
+            // If there's no job or the existing job doesn't match the data we've been passed, escalate to master.
+            if (!result.empty() && result[0][1] == job["data"]){
+                // If we are calling CreateJob, return early, there are no more jobs to create.
+                if (SIEquals(request.methodLine, "CreateJob")) {
```

Comment:
@pecanoro @cead22 Shouldn't we move this `if` statement a few lines up?
Like before the `// Verify unique` line, we don't need to verify that when creating multiple jobs, no?